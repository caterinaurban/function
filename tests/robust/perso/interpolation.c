void main(){
  int x,x0,x1,y0,y1,r;
  y0 = 1;
  y1 = 7; 
  
  int bug ; 



  
  bug = 0 ; 
  if (x0 <= x && x <= x1 && x0 < x1 && y0 <= y1) {
    r = y0 + ((uint64_t) (x-x0) * (y1-y0) / (x1-x0));
    if (!(y0 <= r && r <= y1)) bug = 1 ;
  }

}





