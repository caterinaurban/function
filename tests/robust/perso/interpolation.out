./main.exe
tests/robust/perso/interpolation.c
-domain
polyhedra
-ordinals
3
-robust
tests/robust/perso/interpolation.txt

Abstract Syntax:
[ 1:]
void main( ):
[ 2:] $4{y0} := 1
[ 3:] $5{y1} := 7
[ 4:] $7{bug} := 0
[ 5:] if ( (($2{x0} <= $1{x} && $1{x} <= $3{x1}) && $2{x0} < $3{x1}) && $4{y0} <= $5{y1} ) then
[ 6:]   $6{r} := ?
[ 7:]   if ( !($4{y0} <= $6{r} && $6{r} <= $5{y1}) ) then
[ 8:]     $7{bug} := 1
[ 9:]
        else
[10:]
        endif
[11:]
      else
[12:]
      endif
[13:]

Property: $7{bug} <= 1 && $7{bug} >= 1
[ 2:]:

NODE $7{bug} >= 2
 NODE $2{x0}-$3{x1} >= 0
  NODE $1{x}-$2{x0} >= 0
   NODE $1{x}-$3{x1} >= 1
    LEAF bottom
     LEAF bottom
    NODE $1{x}-$3{x1} >= 1
     LEAF bottom
      LEAF bottom
   NODE $1{x}-$2{x0} >= 0
    NODE $1{x}-$3{x1} >= 1
     LEAF bottom
      LEAF bottom
     LEAF bottom
  NODE $7{bug} >= 1
   NODE $2{x0}-$3{x1} >= 0
    NODE $1{x}-$2{x0} >= 0
     NODE $1{x}-$3{x1} >= 1
      LEAF 0.
       LEAF 0.
      NODE $1{x}-$3{x1} >= 1
       LEAF 0.
        LEAF 0.
     NODE $1{x}-$2{x0} >= 0
      NODE $1{x}-$3{x1} >= 1
       LEAF 0.
        LEAF 0.
       LEAF 0.
    NODE $2{x0}-$3{x1} >= 0
     NODE $1{x}-$2{x0} >= 0
      NODE $1{x}-$3{x1} >= 1
       LEAF bottom
        LEAF bottom
       NODE $1{x}-$3{x1} >= 1
        LEAF bottom
         LEAF bottom
      NODE $1{x}-$2{x0} >= 0
       NODE $1{x}-$3{x1} >= 1
        LEAF bottom
         LEAF bottom
        LEAF bottom
 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$3{x1}-$2{x0}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                                                  $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$3{x1}-$2{x0}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                                           $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$3{x1}-$2{x0}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                                           $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $3{x1}-$2{x0}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                                    $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$2{x0}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                                           $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$2{x0}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                                    $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$2{x0}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                                    $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $2{x0}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                             $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$3{x1}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                                           $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$3{x1}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                                    $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$3{x1}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                                    $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $3{x1}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                             $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                                    $7-1=0; $2-$3>=0|]
[|
                                                    $7-1=0; -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                             $7-1=0; $2-$3>=0|]
[|$7-1=0;
                                                                  -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$1{x}-$6{r}-
  -- constraints   --  
[|
                                             $7-1=0; $2-$3>=0|]
[|$7-1=0;
                                                                  -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $1{x}-$6{r}-
  -- constraints   --  
[|$7-1=0;
                                                                $2-$3>=0|]
[|
                                      $7-1=0; -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$3{x1}-$2{x0}-$6{r}-
  -- constraints   --  
[|
                                                            $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$3{x1}-$2{x0}-$6{r}-
  -- constraints   --  
[|
                                                     $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$3{x1}-$2{x0}-$6{r}-
  -- constraints   --  
[|
                                                     $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $3{x1}-$2{x0}-$6{r}-
  -- constraints   --  
[|
                                              $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$2{x0}-$6{r}-
  -- constraints   --  
[|
                                                     $7-1=0; $1-$3-1>=0|]
[|
                                                     $7-1=0; -$1+$3>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$2{x0}-$6{r}-
  -- constraints   --  
[|
                                              $7-1=0; $1-$3-1>=0|]
[|
                                              $7-1=0; -$1+$3>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$2{x0}-$6{r}-
  -- constraints   --  
[|
                                              $7-1=0; $1-$3-1>=0|]
[|
                                              $7-1=0; -$1+$3>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $2{x0}-$6{r}-
  -- constraints   --  
[|$7-1=0;
                                                                 $1-$3-1>=0|]
[|
                                       $7-1=0; -$1+$3>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$3{x1}-$6{r}-
  -- constraints   --  
[|
                                                     $7-1=0; $1-$2>=0|]
[|
                                                     $7-1=0; -$1+$2-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$3{x1}-$6{r}-
  -- constraints   --  
[|
                                              $7-1=0; $1-$2>=0|]
[|$7-1=0;
                                                                   -$1+$2-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$3{x1}-$6{r}-
  -- constraints   --  
[|
                                              $7-1=0; $1-$2>=0|]
[|$7-1=0;
                                                                   -$1+$2-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $3{x1}-$6{r}-
  -- constraints   --  
[|$7-1=0;
                                                                 $1-$2>=0|]
[|
                                       $7-1=0; -$1+$2-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$6{r}-
  -- constraints   --  
[|
                                              $7-1=0; $2-$3>=0; $1-$2>=0;
                                              $1-$3-1>=0|]
[|-$1+$3=0;
                                                             -$1+$2=0; 
                                                             $7-1=0|]
[|
                                              $7-1=0; -$1+$2-1>=0; $1-$3-1>=0|]
[|
                                              $7-1=0; -$1+$3>=0; -$1+$2-1>=0;
                                              $2-$3>=0|]
[|$7-1=0;
                                                           -$2+$3-1>=0;
                                                           $1-$3-1>=0|]
[|
                                              $7-1=0; -$1+$3>=0; -$2+$3-1>=0;
                                              $1-$2>=0|]
[|$7-1=0;
                                                           -$1+$2-1>=0;
                                                           -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$6{r}-
  -- constraints   --  
[|$7-1=0;
                                                                 $2-$3>=0;
                                                                 $1-$2>=0;
                                                                 $1-$3-1>=0|]
[|
                                       -$1+$3=0; -$1+$2=0; $7-1=0|]
[|
                                       $7-1=0; -$1+$2-1>=0; $1-$3-1>=0|]
[|
                                       $7-1=0; -$1+$3>=0; -$1+$2-1>=0;
                                       $2-$3>=0|]
[|$7-1=0; -$2+$3-1>=0;
                                                    $1-$3-1>=0|]
[|$7-1=0;
                                                                   -$1+$3>=0;
                                                                   -$2+$3-1>=0;
                                                                   $1-$2>=0|]
[|
                                       $7-1=0; -$1+$2-1>=0; -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$6{r}-
  -- constraints   --  
[|$7-1=0;
                                                                 $2-$3>=0;
                                                                 $1-$2>=0;
                                                                 $1-$3-1>=0|]
[|
                                       -$1+$3=0; -$1+$2=0; $7-1=0|]
[|
                                       $7-1=0; -$1+$2-1>=0; $1-$3-1>=0|]
[|
                                       $7-1=0; -$1+$3>=0; -$1+$2-1>=0;
                                       $2-$3>=0|]
[|$7-1=0; -$2+$3-1>=0;
                                                    $1-$3-1>=0|]
[|$7-1=0;
                                                                   -$1+$3>=0;
                                                                   -$2+$3-1>=0;
                                                                   $1-$2>=0|]
[|
                                       $7-1=0; -$1+$2-1>=0; -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $6{r}-
  -- constraints   --  
[|$7-1=0; $2-$3>=0;
                                                          $1-$2>=0;
                                                          $1-$3-1>=0|]
[|
                                -$1+$3=0; -$1+$2=0; $7-1=0|]
[|$7-1=0;
                                                               -$1+$2-1>=0;
                                                               $1-$3-1>=0|]
[|
                                $7-1=0; -$1+$3>=0; -$1+$2-1>=0; $2-$3>=0|]
[|
                                $7-1=0; -$2+$3-1>=0; $1-$3-1>=0|]
[|$7-1=0;
                                                                    -$1+$3>=0;
                                                                    -$2+$3-1>=0;
                                                                    $1-$2>=0|]
[|
                                $7-1=0; -$1+$2-1>=0; -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$3{x1}-$2{x0}-$1{x}-
  -- constraints   --  
[|
                                                            $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$3{x1}-$2{x0}-$1{x}-
  -- constraints   --  
[|
                                                     $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$3{x1}-$2{x0}-$1{x}-
  -- constraints   --  
[|
                                                     $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $3{x1}-$2{x0}-$1{x}-
  -- constraints   --  
[|
                                              $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$2{x0}-$1{x}-
  -- constraints   --  
[|
                                                     $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$2{x0}-$1{x}-
  -- constraints   --  
[|
                                              $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$2{x0}-$1{x}-
  -- constraints   --  
[|
                                              $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $2{x0}-$1{x}-
  -- constraints   --  
[|$7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$3{x1}-$1{x}-
  -- constraints   --  
[|
                                                     $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$3{x1}-$1{x}-
  -- constraints   --  
[|
                                              $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$3{x1}-$1{x}-
  -- constraints   --  
[|
                                              $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $3{x1}-$1{x}-
  -- constraints   --  
[|$7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$1{x}-
  -- constraints   --  
[|
                                              $7-1=0; $2-$3>=0|]
[|$7-1=0;
                                                                   -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$1{x}-
  -- constraints   --  
[|$7-1=0;
                                                                 $2-$3>=0|]
[|
                                       $7-1=0; -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$1{x}-
  -- constraints   --  
[|$7-1=0;
                                                                 $2-$3>=0|]
[|
                                       $7-1=0; -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $1{x}-
  -- constraints   --  
[|$7-1=0; $2-$3>=0|]
[|
                                $7-1=0; -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$3{x1}-$2{x0}-
  -- constraints   --  
[|
                                                      $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$3{x1}-$2{x0}-
  -- constraints   --  
[|
                                               $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$3{x1}-$2{x0}-
  -- constraints   --  
[|
                                               $7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $3{x1}-$2{x0}-
  -- constraints   --  
[|$7-1=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$2{x0}-
  -- constraints   --  
[|
                                               $7-1=0; $1-$3-1>=0|]
[|
                                               $7-1=0; -$1+$3>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$2{x0}-
  -- constraints   --  
[|$7-1=0;
                                                                  $1-$3-1>=0|]
[|
                                        $7-1=0; -$1+$3>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$2{x0}-
  -- constraints   --  
[|$7-1=0;
                                                                  $1-$3-1>=0|]
[|
                                        $7-1=0; -$1+$3>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $2{x0}-
  -- constraints   --  
[|$7-1=0; $1-$3-1>=0|]
[|
                                 $7-1=0; -$1+$3>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-$3{x1}-
  -- constraints   --  
[|
                                               $7-1=0; $1-$2>=0|]
[|$7-1=0;
                                                                    -$1+$2-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-$3{x1}-
  -- constraints   --  
[|$7-1=0;
                                                                  $1-$2>=0|]
[|
                                        $7-1=0; -$1+$2-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$3{x1}-
  -- constraints   --  
[|$7-1=0;
                                                                  $1-$2>=0|]
[|
                                        $7-1=0; -$1+$2-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $3{x1}-
  -- constraints   --  
[|$7-1=0; $1-$2>=0|]
[|
                                 $7-1=0; -$1+$2-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-$4{y0}-
  -- constraints   --  
[|$7-1=0;
                                                                  $2-$3>=0;
                                                                  $1-$2>=0;
                                                                  $1-$3-1>=0|]
[|
                                        -$1+$3=0; -$1+$2=0; $7-1=0|]
[|
                                        $7-1=0; -$1+$2-1>=0; $1-$3-1>=0|]
[|
                                        $7-1=0; -$1+$3>=0; -$1+$2-1>=0;
                                        $2-$3>=0|]
[|$7-1=0; -$2+$3-1>=0;
                                                     $1-$3-1>=0|]
[|$7-1=0;
                                                                    -$1+$3>=0;
                                                                    -$2+$3-1>=0;
                                                                    $1-$2>=0|]
[|
                                        $7-1=0; -$1+$2-1>=0; -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $4{y0}-
  -- constraints   --  
[|$7-1=0; $2-$3>=0;
                                                           $1-$2>=0;
                                                           $1-$3-1>=0|]
[|
                                 -$1+$3=0; -$1+$2=0; $7-1=0|]
[|$7-1=0;
                                                                -$1+$2-1>=0;
                                                                $1-$3-1>=0|]
[|
                                 $7-1=0; -$1+$3>=0; -$1+$2-1>=0; $2-$3>=0|]
[|
                                 $7-1=0; -$2+$3-1>=0; $1-$3-1>=0|]
[|
                                 $7-1=0; -$1+$3>=0; -$2+$3-1>=0; $1-$2>=0|]
[|
                                 $7-1=0; -$1+$2-1>=0; -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

 --   uncontrolled  -- 
 $5{y1}-
  -- constraints   --  
[|$7-1=0; $2-$3>=0;
                                                           $1-$2>=0;
                                                           $1-$3-1>=0|]
[|
                                 -$1+$3=0; -$1+$2=0; $7-1=0|]
[|$7-1=0;
                                                                -$1+$2-1>=0;
                                                                $1-$3-1>=0|]
[|
                                 $7-1=0; -$1+$3>=0; -$1+$2-1>=0; $2-$3>=0|]
[|
                                 $7-1=0; -$2+$3-1>=0; $1-$3-1>=0|]
[|
                                 $7-1=0; -$1+$3>=0; -$2+$3-1>=0; $1-$2>=0|]
[|
                                 $7-1=0; -$1+$2-1>=0; -$2+$3-1>=0|]

 -- Join constraint --  [|$7-1=0|]

simply reachable ?: true 


Analysis Result: UNKNOWN

Done.
